# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/xnnpack/hotfix-cpuinfo.patch
# Copyright (C) 2026 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

--- XNNPACK-124d961219d5486cda611908be55f9b5ba82ac8c/CMakeLists.txt.vanilla	2026-08-07 19:41:59.637327217 +0300
+++ XNNPACK-124d961219d5486cda611908be55f9b5ba82ac8c/CMakeLists.txt	2026-08-07 19:42:34.839364150 +0300
@@ -1262,15 +1262,6 @@
 
 # ---[ Configure cpuinfo
 IF(NOT TARGET cpuinfo)
-  IF(NOT XNNPACK_USE_SYSTEM_LIBS)
-    SET(CPUINFO_BUILD_TOOLS OFF CACHE BOOL "")
-    SET(CPUINFO_BUILD_UNIT_TESTS OFF CACHE BOOL "")
-    SET(CPUINFO_BUILD_MOCK_TESTS OFF CACHE BOOL "")
-    SET(CPUINFO_BUILD_BENCHMARKS OFF CACHE BOOL "")
-    ADD_SUBDIRECTORY(
-      "${CPUINFO_SOURCE_DIR}"
-      "${CMAKE_BINARY_DIR}/cpuinfo")
-  ELSE()
     ADD_LIBRARY(cpuinfo SHARED IMPORTED)
     FIND_LIBRARY(CPUINFO_LIBRARY cpuinfo PATHS "${CPUINFO_SOURCE_DIR}/lib64")
     IF(NOT CPUINFO_LIBRARY)
@@ -1279,7 +1270,6 @@
     TARGET_INCLUDE_DIRECTORIES(cpuinfo INTERFACE "${CPUINFO_SOURCE_DIR}/include")
     SET_PROPERTY(TARGET cpuinfo PROPERTY IMPORTED_LOCATION "${CPUINFO_LIBRARY}")
     SET_PROPERTY(TARGET cpuinfo PROPERTY IMPORTED_IMPLIB "${CPUINFO_LIBRARY}")
-  ENDIF()
 ENDIF()
 IF(XNNPACK_BUILD_LIBRARY)
   TARGET_LINK_LIBRARIES(xnnpack-hardware-config PRIVATE cpuinfo)

